flink-table/flink-table-common/src/test/java/org/apache/flink/table/types/LegacyTypeInfoDataTypeConverterTest.java

```diff
@@ -93,6 +98,20 @@
 						.bridgedTo(float[][].class)
 				},
 
+				{
+					BasicArrayTypeInfo.getInfoFor(String[].class),
+					new AtomicDataType(
+						new LegacyTypeInformationType<>(
+							LogicalTypeRoot.ARRAY,
+							Types.OBJECT_ARRAY(Types.STRING)))
```

Comment:
nit: Could we be explicit here? I would not notice that `Types.OBJECT_ARRAY` produces `BasicArrayTypeInfo` for `String` if I didn't know the method previously.